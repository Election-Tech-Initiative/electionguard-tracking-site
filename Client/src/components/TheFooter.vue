<template>
    <footer class="footer">
        <div class="container">
            <div class="content">
                {{ $t('footerTitle', { location }) }}
                <span class="separator" />
                <span class="is-inline-block">
                {{ $t('footerCopyright') }}
                </span>
                <span class="separator" />
                <router-link 
                    class="is-inline-block has-text-weight-semibold"
                    :to="linkBallotTracking"
                >
                {{ $t('ballotTracking') }}
                </router-link>               
                <span class="separator" />
                <router-link 
                    class="is-inline-block has-text-weight-semibold"
                    :to="linkElectionResults"
                >
                {{ $t('electionResults') }}
                </router-link>
                <span class="separator" />
                <LanguageToggle />
            </div>
        </div> 
    </footer>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import LanguageToggle from '@/components/LanguageToggle.vue';

@Component({
    components: { LanguageToggle }
})
export default class TheFooter extends Vue {
  public get linkBallotTracking(): string {
      return '/' + this.langQuery;
  }

  public get linkElectionResults(): string {
      return '/election-results' + this.langQuery;
  }

  public get langQuery(): string {
      const lang = this.$route.query.lang;
      return lang ? `?lang=${lang}` : '';
  }

  public get location(): string {
    if (this.appSettings.locationDescription2) {
      return `${this.appSettings.locationDescription}, ${this.appSettings.locationDescription2}`;
    }
    return this.appSettings.locationDescription;
  }
}
</script>

<style lang="scss" scoped>
    .footer {
        background-color: transparent;
        font-size: 0.9rem;

        @include mobile {
            padding-top: 2rem;
            padding-bottom: 2rem;
            font-size: 0.8rem;
        }

        a {
            color: $text;

            &.router-link-exact-active {
                color: $primary;
            }
        }

        .separator {
            &::before {
                content: '\00b7';
                padding: 0 0.5rem;
            }
        }

        .language-toggle {
            display: inline-block;
        }
    }
</style>

