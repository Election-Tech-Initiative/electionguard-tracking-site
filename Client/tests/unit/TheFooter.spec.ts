import { mount, RouterLinkStub } from '@vue/test-utils';

import TheFooter from '@/components/TheFooter.vue';

jest.mock('@/components/LanguageToggle.vue', () => ({ default: jest.fn() }));

describe('TheFooter.vue', () => {
    it('renders', () => {
        const wrapper = mount(TheFooter, {
            mocks: {
                appSettings: {
                    locationDescription: 'Mocked Location',
                    locationDescription2: 'Mocked Location 2',
                },
                $route: {
                    query: {
                        lang: 'mockedLang',
                    },
                },
                $t: (msg: string): string => msg
            },
            stubs: {
                LanguageToggle: true,
                RouterLink: RouterLinkStub,
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
