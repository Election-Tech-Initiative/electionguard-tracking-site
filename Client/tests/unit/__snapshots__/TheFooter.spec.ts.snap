// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheFooter.vue renders 1`] = `
<footer
  class="footer"
>
  <div
    class="container"
  >
    <div
      class="content"
    >
      
            footerTitle
            
      <span
        class="separator"
      />
       
      <span
        class="is-inline-block"
      >
        
            footerCopyright
            
      </span>
       
      <span
        class="separator"
      />
       
      <a
        class="is-inline-block has-text-weight-semibold"
      >
        
            ballotTracking
            
      </a>
       
      <span
        class="separator"
      />
       
      <a
        class="is-inline-block has-text-weight-semibold"
      >
        
            electionResults
            
      </a>
       
      <span
        class="separator"
      />
       
      <languagetoggle-stub />
    </div>
  </div>
</footer>
`;
